src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -198,6 +198,31 @@ private void CreateNodeCaches()
                     MethodEntrypoint(method.GetCanonMethodTarget(CanonicalFormKind.Specific)));
             });
 
+            _nativeLayoutTypeSignatures = new NodeCache<TypeDesc, NativeLayoutTypeSignatureVertexNode>(type =>
```

Comment:
Could you stash these away into e.g. `NodeFactory.NativeLayout.cs`, same as we have `NodeFactory.GenericLookups.cs`? Make a nested class to fold them away in IntelliSense, etc.

This group will only grow longer and doing this will help in maintainability.